tests/unittests/documents.py

```diff
@@ -0,0 +1,36 @@
+import unittest
+from mongoengine import EmbeddedDocument
+from mongoengine.errors import ValidationError
+from fiftyone.core.fields import EmbeddedDocumentListField
+from fiftyone.core.odm.dataset import SampleFieldDocument
+
+
+class MockEmbeddedDocument(EmbeddedDocument):
+    pass
+
+
+class TestEmbeddedDocumentListField(unittest.TestCase):
+    def test_validate(self):
```

Comment:
@minhtuev can you also add an integration test here to clarify why we're testing this in the first place?

```py
dataset = fo.Dataset()
dataset._doc.sample_fields[0] = None

# sample field docs cannot be None
with self.assertRaises(ValidationError):
    dataset.save()
```

Reply:
Done ✅